From 866eb40fe4fc5c5bd8b9fca088f3da7831b0b054 Mon Sep 17 00:00:00 2001
From: Michael Pauzner <mpauzner@habana.ai>
Date: Mon, 05 May 2025 16:40:25 +0300
Subject: [PATCH] [SW-221409] g3: serdes: early lfs image creation

Move creation of serdes partition image to an earlier
build stage. Because several itb files need the image,
including gaudi3-boot-fit.itb.

Change-Id: I002d6c61842727e3f3911c62e57f5b9de7759438
Signed-off-by: Michael Pauzner <mpauzner@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index 1cae988..a0b2792 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -522,6 +522,4 @@
 		source ${KEY_BASE}/spi-flash-layout-validator.sh
 		validate_spi_flash_layout
 	fi
-
-	prep_image_serdes_config
 popd
diff --git a/board/habanalabs/fs1_arc/post-image-spi.sh b/board/habanalabs/fs1_arc/post-image-spi.sh
index 5eca164..d98e307 100755
--- a/board/habanalabs/fs1_arc/post-image-spi.sh
+++ b/board/habanalabs/fs1_arc/post-image-spi.sh
@@ -703,8 +703,3 @@
 	# pad arcboot rom binary to size of ROM memory
 	dd status=none conv=sync bs=128K if=${ARCBOOT_BINFILE_ORIG} > ${ARCBOOT_BINFILE}
 }
-
-function prep_image_serdes_config {
-	# dummy function, not supported for fs1
-	return
-}
\ No newline at end of file
diff --git a/board/habanalabs/gaudi2_arc/post-image-spi.sh b/board/habanalabs/gaudi2_arc/post-image-spi.sh
index 2bae0f0..6358fd6 100755
--- a/board/habanalabs/gaudi2_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi2_arc/post-image-spi.sh
@@ -505,8 +505,3 @@
 	# dummy function, not supported for gaudi2
 	return
 }
-
-function prep_image_serdes_config {
-	# dummy function, not supported for gaudi2
-	return
-}
\ No newline at end of file
diff --git a/board/habanalabs/gaudi3/post-image-fit.sh b/board/habanalabs/gaudi3/post-image-fit.sh
index a54c8f8..a602cdd 100644
--- a/board/habanalabs/gaudi3/post-image-fit.sh
+++ b/board/habanalabs/gaudi3/post-image-fit.sh
@@ -70,6 +70,8 @@
 
 	prep_eth_phy_fw_binary ${ETH_PHY_FW_HEX_FILE} ${ETH_PHY_FW_BIN_FILE}
 
+	prep_image_serdes_config
+
 	return
 }
 
@@ -107,3 +109,15 @@
 ITS_SED_PARAMS+="s/@MGMT_STARTADDR_LO@/${MGMT_STARTADDR_LO}/g;"
 ITS_SED_PARAMS+="s/@EVENT_STARTADDR_HI@/${ARC_EVENT_ICCM_HI}/g;"
 ITS_SED_PARAMS+="s/@EVENT_STARTADDR_LO@/${EVENT_STARTADDR_LO}/g;"
+
+SERDES_LFS_BIN_FILE_NAME="gaudi3-serdes-config.bin"
+SERDES_LFS_FILE="${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${SERDES_LFS_BIN_FILE_NAME}"
+SERDES_LFS_JSON_FILE="${BR2_EXTERNAL}/scripts/serdes/gaudi3_serdes_config_flash_memory.json"
+
+function prep_image_serdes_config {
+	# Generate the serdes LFS config bin file
+	${BR2_EXTERNAL}/scripts/serdes/serdes_generate_littlefs_flash_memory.py 	\
+							-o ${SERDES_LFS_FILE}		\
+							-j ${SERDES_LFS_JSON_FILE}
+	return
+}
diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index bcdf092..1da72b7 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -798,15 +798,3 @@
 	# pad arcboot rom binary to size of ROM memory
 	dd status=none conv=sync bs=128K if=${BINS_DIR}/${ARCBOOT_BINFILE_ORIG} > ${BINS_PROD_DIR}/${ARCBOOT_BINFILE}
 }
-
-SERDES_LFS_BIN_FILE_NAME="gaudi3-serdes-config.bin"
-SERDES_LFS_FILE="${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${SERDES_LFS_BIN_FILE_NAME}"
-SERDES_LFS_JSON_FILE="${BR2_EXTERNAL}/scripts/serdes/gaudi3_serdes_config_flash_memory.json"
-
-function prep_image_serdes_config {
-	# Generate the serdes LFS config bin file
-	${BR2_EXTERNAL}/scripts/serdes/serdes_generate_littlefs_flash_memory.py 	\
-							-o ${SERDES_LFS_FILE}		\
-							-j ${SERDES_LFS_JSON_FILE}
-	return
-}
